cli/compose/loader/loader.go

```diff
@@ -427,9 +427,14 @@ func loadVolumes(source types.Dict) (map[string]types.VolumeConfig, error) {
 		return volumes, err
 	}
 	for name, volume := range volumes {
-		if volume.External.External && volume.External.Name == "" {
-			volume.External.Name = name
-			volumes[name] = volume
+		if volume.External.External {
+			if len(volume.DriverOpts) > 0 {
+				return nil, fmt.Errorf("both 'external' and options are specified for volume %q. Only one is allowed", name)
+			}
```

Comment:
https://docs.docker.com/compose/compose-file/#/external it's not specified — @dnephin @shin-